eventmachine.gemspec

```diff
@@ -12,7 +12,12 @@ Gem::Specification.new do |s|
   s.files = `git ls-files README.md CHANGELOG.md GNU LICENSE rakelib ext java lib docs`.split
   s.test_files = `git ls-files tests examples`.split
 
-  s.extensions = ["ext/extconf.rb", "ext/fastfilereader/extconf.rb"]
+  if RUBY_PLATFORM =~ /java/
+    s.files << "lib/rubyeventmachine.jar"
+    s.platform = 'java'
+  else
+    s.extensions = ["ext/extconf.rb", "ext/fastfilereader/extconf.rb"]
+  end
```

Comment:
Setting the platform and adding the jar file will build a 'java' gem that will be preferentially searched for installed if you're on a java platform when it's available on rubygems.

Segregating out the extensions will prevent jruby from trying to build those extensions (which it can't do) so that the gem can be installed.